src/porepy/applications/md_grids/mdg_library.py

```diff
@@ -172,3 +173,51 @@ def benchmark_regular_2d(
     if is_coarse:
         pp.coarsening.coarsen(mdg, "by_volume")
     return mdg, fracture_network
+
+
+def benchmark_3d_case_3(
+    refinement_level: Literal[0, 1, 2, 3] = 0,
+) -> tuple[pp.MixedDimensionalGrid, pp.FractureNetwork3d]:
+    """
+    Create a mixed-dimensional grid for the geometry of case 3 from [1].
+
+    Note:
+        The mixed-dimensional grid is created by reading a `geo` file, so there is no
+        direct way of prescribing meshing arguments.
+
+    Reference:
+        [1] Berre, I., Boon, W. M., Flemisch, B., Fumagalli, A., Gläser, D., Keilegavlen,
+        E., ... & Zulian, P. (2021). Verification benchmarks for single-phase flow in
+        three-dimensional fractured porous media. Advances in Water Resources, 147,
+        103759.
+
+    Parameters:
+        refinement_level: An integer denoting the level of refinement. Use `0` to
+            generate a mixed-dimensional grid with approximately 30K 3D cells,
+            `1` for 140K 3D cells, `2` for 350K 3D cells, and `3` for 500K 3D cells.
+            Default is `0`.
+
+    Returns:
+        Mixed-dimensional grid for the given refinement level, and the fracture network.
```

Comment:
Strictly speaking, this does not follow the doc style for multiple returns. I'm fine with not fixing now, but wanted to draw your attention to it.